code_map.py

```diff
@@ -352,30 +435,55 @@ def run(self, edit, **args):
 
         # generate new map
         source = args['source']
-        map = ""
         map_syntax = py_syntax
 
         try:
-            if using_universal_mapper:
-                (generate, syntax) = code_map_generator.get_mapper(source)
+            # it's the id of the temporary view
+            if type(source) != str:
+                for v in sublime.active_window().views():
+                    if v.id() == source:
+                        (map, map_syntax) = code_map_generator.view_to_map(v)
 
-                if type(generate) is str:
-                    map = generate
-                else:
-                    map = generate(source)
+            else:
+                # use temp map that has been generated, then delete it
+                map = Generated_Map
+                Generated_Map = None
+                if not map:
+                    # probably not necessary but to be sure
+                    map = code_map_generator.get_mapper(source)
 
-                map_syntax = syntax
+                '''using_universal_mapper variable is set in get_mapper, when
+                it comes back here the script knows already where to go'''
 
-            else:
-                (generate, syntax) = code_map_generator.get_mapper(source)
-                map = generate(source)
-                map_syntax = syntax
+                # refresh map -> get_mapper -> sets variable -> then here
+
+                '''this detour seems necessary to me because the variable
+                can't be set here, it can only be set where the file type is
+                being evaluated, but how the result must be processed is
+                defined here'''
+
+                if using_universal_mapper:
+                    (map, map_syntax) = map
+
```

Comment:
When it comes here, it has gone already through get_mapper(), where variables have been set